CodingChallenges/CC_092_xor/Processing/CC_092_xor/Neural_Network.pde

```diff
@@ -0,0 +1,207 @@
+float sigmoid(float x)
+{
+  return 1 / (1+(float)Math.exp(-x));
+}
+
+float dsigmoid(float y)
+{
+  return y * (1-y);
+}
+
+float tanh(float x)
+{
+  float y = (float) Math.tanh(x);
+  return y;
+}
+
+float dtanh(float x) 
+{
+  float y = 1 / (pow((float) Math.cosh(x), 2));
+  return y;
+}
+
+class NeuralNetwork
+{
+  int 
+    inputNodes, 
+    hiddenNodes, 
+    outputNodes;
+
+  float LearningRate = .1;
+
+  Matrix 

```

Comment:
It's long winded, but the style I tend to use (and the variable names should be lower case):

```java
  Matrix ihWeights;
  Matrix hoWeights;
  Matrix hBias;
  Matrix oBias;
  Matrix input;
  Matrix hidden;
  Matrix output;
```